test/Distributed/Runtime/distributed_actor_func_calls_remoteCall_through_generic_and_inner.swift

```diff
@@ -0,0 +1,98 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend-emit-module -emit-module-path %t/FakeDistributedActorSystems.swiftmodule -module-name FakeDistributedActorSystems -disable-availability-checking %S/../Inputs/FakeDistributedActorSystems.swift
+// RUN: %target-build-swift -module-name main  -Xfrontend -disable-availability-checking -j2 -parse-as-library -I %t %s %S/../Inputs/FakeDistributedActorSystems.swift -o %t/a.out
+// RUN: %target-run %t/a.out | %FileCheck %s --color
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: distributed
+
+// rdar://76038845
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: back_deployment_runtime
+
+// FIXME(distributed): Distributed actors currently have some issues on windows, isRemote always returns false. rdar://82593574
+// UNSUPPORTED: OS=windows-msvc
+
+import Distributed
+import FakeDistributedActorSystems
+
+typealias DefaultDistributedActorSystem = FakeRoundtripActorSystem
+
+protocol Greeting: DistributedActor {
+  distributed func greeting() -> String
+  distributed func greetingAsyncThrows() async throws -> String
+}
+
+extension Greeting {
+  func greetLocal(name: String) async throws {
+    try await print("\(greetingAsyncThrows()), \(name)!") // requirement is async throws, things work
+  }
+
+  // FIXME(distributed): won't work today, it will try to treat the greeting() call as async, but can't
+  //  We should understand that greeting is called on self, and not use the thunk witness.
+//  func greetLocal2(name: String) {
+//    print("\(greeting()), \(name)!") // okay, we're on the actor
+//  }
```

Comment:
This is one last remaining case I could not figure out how to make work yet;

We have to detect that "the call is on self, therefore, do not substitute the constant for the call with the thunk for the witness". But nowhere where we do that, do we have the expr to base it on... In my other `whenLocal` work I did a bunch of improvements around detecting "when a thing is local" like detecting the base of a call... but we still need expressions.

The question about accepting this PR is:
- is this a big enough improvement, to accept this one missing case (it does cause a crash, and the workaround is to make the `greeting()` requirement async throws), that we'd fix later
- or, is this risky and incomplete in a way that we prefer to not merge it.

This also impacts the mangling conflict issue... although I did not manage to switch us over to always passing the `func + asDistributed()` declref rather than the current `thunk + asDistributed()`, but this PR would enable it... rdar://96179707

Open to any plans about how we should tackle this tbh; we can take it slow with this PR as well, not to risk breaking 5.7.